source/OutfitterPanel.cpp

```diff
@@ -363,15 +362,9 @@ void OutfitterPanel::Buy(bool alreadyOwned)
 		int mapSize = selectedOutfit->Get("map");
 		if(mapSize > 0)
 		{
-			if(!HasMapped(mapSize))
```

Comment:
Why removal of the check if the player already has this distance mapped?

Reply:
It seemed to me like a duplicate check, as `::CanBuy` already does it: https://github.com/endless-sky/endless-sky/blob/master/source/OutfitterPanel.cpp#L316

Are there are flows where `::Buy` is used alone without first going through `::CanBuy`?

Reply:
Consider it like server-side validation on the web. Even if the client-side does it, it's a good idea to verify assumptions before acting.

Reply:
Restored with d2a39cd